testsuite/tests/tsan/waitgroup_stubs.c

```diff
@@ -6,7 +6,7 @@
 #include <caml/tsan.h>
 
 #define MAX_WAITGROUP   8
-#define SPIN_WAIT_MS    10
+#define SPIN_WAIT_US    10
```

Comment:
It is not clear to me whether this code originally intended to sleep for 10ms or 10us, and what is the right value. (10us feels like it may hog CI resources with a busy-waiting loop that is too fast; and I don't think that we rely on the fact that domains exit the "join" barrier close together, but I'm not fully sure) @fabbing, do you remember?

Reply:
`Waitgroup`s are used for a `Domain` to wait for another domain to complete its test, which can take some time.
So the sleep time was deliberately high to avoid wasting too much CPU cycles. 10us seems a very low pause, do you think 10ms was too much?

Reply:
The existing code passes this `SLEEP_MS` value to `usleep`, which epects microseconds, not milliseconds. This PR changes the suffix to `_US` to disambiguate this, but does not change the sleep duration.

Reply:
My understanding from @fabbing's reply (thanks!) is that 10ms were intended, and I would suggest moving to this. (Maybe we could do it as a quick bugfix-only PR first, to see if the behavior change breaks something somewhere. @dustanddreams would you want to try that?)

Reply:
I see no difference in behaviour of the tsan tests with this duration promoted from microseconds to mililseconds.

Reply:
Good to go? Should I add a dash of microseconds?

Reply:
My understanding is that the code was initially written with the intention of sleeping for 10ms, but in fact (both before and after this PR) it sleeps for 10us. I would prefer if the PR would fix the bug (mismatch between intention and implementation) by changing `SPIN_WAIT_*` to wait for 10ms.

Reply:
Done, it's now waiting for 10ms.